lib/nostrum/struct/sticker/pack.ex

```diff
@@ -0,0 +1,93 @@
+defmodule Nostrum.Struct.Sticker.Pack do
+  @moduledoc """
+  Represents an in-built sticker pack on Discord
+  """
+
+  defstruct [
+    :id,
+    :stickers,
+    :name,
+    :sku_id,
+    :cover_sticker_id,
+    :description,
+    :banner_asset_id
+  ]
+
+  alias Nostrum.{Constants, Snowflake, Util}
+  alias Nostrum.Struct.Sticker
+
+  @typedoc """
+  ID of the sticker pack.
+  """
+  @type id :: Snowflake.t()
+
+  @typedoc """
+  A list of stickers contained within the pack.
+  """
+  @type stickers :: [Sticker.t()]
+
+  @typedoc """
+  Name of the pack.
+  """
+  @type name :: String.t()
+
+  @typedoc """
+  SKU ID of the sticker pack.
+  """
+  @type sku_id :: Snowflake.t()
+
+  @typedoc """
+  ID of a sticker contained within the pack that should be the cover.
+  """
+  @type cover_sticker_id :: Sticker.t()
+
+  @typedoc """
+  Marketing description of the sticker pack.
+  """
+  @type description :: String.t()
+
+  @typedoc """
+  Asset ID of the banner for this sticker pack.
+  """
+  @type banner_asset_id :: Snowflake.t()
+
+  @type t :: %__MODULE__{
+          id: id,
+          stickers: stickers,
+          name: name,
+          sku_id: sku_id,
+          cover_sticker_id: cover_sticker_id,
+          description: description,
+          banner_asset_id: banner_asset_id
+        }
+
+  @doc false
+  def to_struct(map) do
+    new =
+      map
+      |> Map.new(fn {k, v} -> {Util.maybe_to_atom(k), v} end)
+      |> Map.update(:stickers, nil, &Util.cast(&1, {:list, {:struct, Sticker}}))
+      |> Map.update(:id, nil, &Util.cast(&1, Snowflake))
+      |> Map.update(:sku_id, nil, &Util.cast(&1, Snowflake))
+      |> Map.update(:cover_sticker_id, nil, &Util.cast(&1, Snowflake))
+      |> Map.update(:banner_asset_id, nil, &Util.cast(&1, Snowflake))
+
+    struct(__MODULE__, new)
+  end
+
+  @doc ~S"""
+  Return the banner pack URL for a given sticker pack.
```

Comment:
What is the banner pack URL for? For stickers it's clear: it returns the sticker as a picture. Or as some strange JSON thing, in case of the lottie format, apparently. But what does a "banner pack URL" contain, especially if we document it as returning a PNG?

Reply:
It's a marketing banner for the platform curated sticker packs, for example: https://cdn.discordapp.com/app-assets/710982414301790216/store/796139192806801428.png?size=1024